File: ProjectTest/OneLang/CSharp_Regen_CSharp/One/Transforms/InferTypesPlugins/NullabilityCheckWithNot.cs
using One.Transforms.InferTypesPlugins.Helpers;
using One.Ast;

namespace One.Transforms.InferTypesPlugins
{
    public class NullabilityCheckWithNot : InferTypesPlugin {
        public NullabilityCheckWithNot(): base("NullabilityCheckWithNot")
        {
            
        }
        
        public override bool canTransform(Expression expr) {
            return expr is UnaryExpression ? ((UnaryExpression)expr).operator_ == "!" : false;
        }
        
        public override Expression transform(Expression expr) {
            var unaryExpr = ((UnaryExpression)expr);
            if (unaryExpr.operator_ == "!") {
                this.main.processExpression(expr);
                var type = unaryExpr.operand.actualType;
                var litTypes = this.main.currentFile.literalTypes;
                if (type is ClassType && ((ClassType)type).decl != litTypes.boolean.decl && ((ClassType)type).decl != litTypes.numeric.decl)
                    return new BinaryExpression(unaryExpr.operand, "==", new NullLiteral());
            }
            
            return null;
        }
    }
}

File: ProjectTest/OneLang/Java/src/main/java/OneLang/One/Transforms/InferTypesPlugins/NullabilityCheckWithNot/NullabilityCheckWithNot.java
public class NullabilityCheckWithNot extends InferTypesPlugin {
    public NullabilityCheckWithNot()
    {
        super("NullabilityCheckWithNot");
        
    }
    
    public Boolean canTransform(Expression expr) {
        return expr instanceof UnaryExpression ? ((UnaryExpression)expr).operator.equals("!") : false;
    }
    
    public Expression transform(Expression expr) {
        var unaryExpr = ((UnaryExpression)expr);
        if (unaryExpr.operator.equals("!")) {
            this.main.processExpression(expr);
            var type = unaryExpr.operand.actualType;
            var litTypes = this.main.currentFile.literalTypes;
            if (type instanceof ClassType && ((ClassType)type).decl != litTypes.boolean_.decl && ((ClassType)type).decl != litTypes.numeric.decl)
                return new BinaryExpression(unaryExpr.operand, "==", new NullLiteral());
        }
        
        return null;
    }
}